commands.go

```diff
@@ -617,6 +610,42 @@ func UploadWALFile(tarUploader *TarUploader, dirArc string, pre *S3Prefix, verif
 	}
 	if err != nil {
 		log.Printf("upload: could not upload '%s'\n", path)
-		log.Fatalf("FATAL %v\n", err)
+		log.Fatalf("FATAL: %v\n", err)
 	}
 }
+
+func checkWALOverwrite(pre *S3Prefix, tarUploader *TarUploader, dirArc string) {
+	archiveReader, exists, err := tryDownloadWALFile(pre,
+		sanitizePath(tarUploader.server+WalPath+filepath.Base(dirArc)+"."+tarUploader.compressor.FileExtension()))
+	if err != nil {
+		log.Fatalf("%+v\n", err)
+	}
+	if exists {
+		archived := &bytes.Buffer{}
+
+		err = decompressWALFile(archiveReader, &bufCloser{archived}, getDecompressorByCompressor(tarUploader.compressor))
+
+		local, err := os.Open(dirArc)
+		defer local.Close()
+
+		localBytes, err := ioutil.ReadAll(local)
+
+		if err != nil {
+			log.Fatalf("FATAL: %+v\n", err)
+		}
+
+		if !bytes.Equal(archived.Bytes(), localBytes) {
+			log.Fatalf("FATAL: WAL file '%s' already archived, contents differ, unable to overwrite\n", dirArc)
+		} else {
+			log.Printf("WARNING: WAL file '%s' already archived, archived content equals\n", dirArc)
+		}
+	}
```

Comment:
This function looks fine design-wise, the only thing that might potentially be a worry is how much slower this makes the uploads.

Reply:
well, uploads are reversed: you download and compare. I didn't benchmark though, I think it is not a normal situation when you try to overwrite WAL, even if with the same data.